### PR TITLE
adding MR1K3_In veto into TMO motion PLC

### PR DESCRIPTION
**Pull request by tongju12**
Adding TXI mirror MR1K3 veto into TMO motion PLCs

Adding one more veto device
## Motivation and Context

https://jira.slac.stanford.edu/browse/ECS-5642
TXI takes beam, it will required MR1K1 IN and possible ST3K4 OUT(not decided). If ST3K4 OUT, then all TMO devices have possibility to trip TXI beam if scientists try to move anything.
So adding MR1K3_IN as veto device for TMO motion, it will help TXI if ST3K4 OUT during TXI beam time.

## How Has This Been Tested?

Tested with Maggie to force PMPS MR1K3 and ST3K4 OUT, move IM4K4 and works as expected. 

## Where Has This Been Documented?

https://jira.slac.stanford.edu/browse/ECS-5642

## Pre-merge checklist
- [ ] Code works interactively
- [ ] Code contains descriptive comments
- [ ] Test suite passes locally
- [ ] Libraries are set to fixed versions and not ``Always Newest``
- [ ] Code committed with ``pre-commit`` (alternatively ``pre-commit run --all-files``)

### CODE REVIEW

**Comment by ZLLentz** [comment]
Hi Tong, can you link the related jira in the description under the "motivation and context" header? This will help us understand changes when we look back on them later.

File: plc-tmo-motion/tmo_motion/POUs/PRG_3_PMPS_POST.TcPOU [hunk 1]
```diff
@@ -17,19 +18,22 @@ END_VAR]]></Declaration>
 bST4K4_Veto := PMPS_GVL.stCurrentBeamParameters.aVetoDevices[PMPS.K_Stopper.ST4K4];
 bM1K1Veto := NOT PMPS_GVL.stCurrentBeamParameters.aVetoDevices[PMPS.K_Stopper.MR1K1_OUT]
                 AND PMPS_GVL.stCurrentBeamParameters.aVetoDevices[PMPS.K_Stopper.MR1K1_IN];
+bM1K3Veto := NOT PMPS_GVL.stCurrentBeamParameters.aVetoDevices[PMPS.K_Stopper.MR1K3_OUT]
+                AND PMPS_GVL.stCurrentBeamParameters.aVetoDevices[PMPS.K_Stopper.MR1K3_IN];
 
+// Adding TXI MR1K3 veto in
```
Comment (ZLLentz) [inline plc-tmo-motion/tmo_motion/POUs/PRG_3_PMPS_POST.TcPOU:24]: Small nitpick: this comment on line 24 and the copy/paste of it on line 36 will be confusing when we look back at this in about a year. Comments should describe what is here, not what is changing.

**Review by ZLLentz — COMMENTED** [review]
Implementation looks correct to me, but I can't confirm because no linked jira for context

File: plc-tmo-motion/tmo_motion/POUs/PRG_3_PMPS_POST.TcPOU (hunk 1 above)
Comment (tongju12) [reply plc-tmo-motion/tmo_motion/POUs/PRG_3_PMPS_POST.TcPOU:24]: Thanks, I will modify on my next PR.

**Review by ZLLentz — APPROVED** [review]
Makes tons of sense with the added context! Thank you!

File: plc-tmo-motion/tmo_motion/POUs/PRG_3_PMPS_POST.TcPOU [hunk 2]
```diff
@@ -17,19 +18,22 @@ END_VAR]]></Declaration>
...
 fbArbiterIO(
     i_bVeto:=bST3K4_Veto,
```
Comment (ZLLentz) [inline plc-tmo-motion/tmo_motion/POUs/PRG_3_PMPS_POST.TcPOU:26]: @tongju12, this review comment is a little bit late, but is this arbiter supposed to be vetoed by MR1K1 and MR1K3 too?

Comment (ZLLentz) [reply plc-tmo-motion/tmo_motion/POUs/PRG_3_PMPS_POST.TcPOU:26]: @ghalym Tong and I are both talking about this offline and are a little confused, can you remind us how these vetoes are supposed to be set up? What is vetoed in fbArbiterIO vs fbVetoArbiter?

**Pull request merged**